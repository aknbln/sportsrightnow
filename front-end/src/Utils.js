export function sumTests(teamData){
    let sum = 0
    teamData.forEach((member) => {
      sum += member.unit_tests
    })
  
    return sum
  }

export function inchToFeet(inches){
  let feet = Math.floor(inches / 12)
  let remainder = inches % 12

  return feet + "\' " + remainder  + "\'"
}