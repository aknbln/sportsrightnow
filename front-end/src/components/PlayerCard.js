import React from "react";

import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import { Navigate, useNavigate, useSearchParams } from "react-router-dom";
import { Highlight } from "../Utils";

const PlayerCard = (props) => {
  const { name, headshot, team, league, college, jersey, id} = props.playerData;
  const search = [props.q]
  const navigate = useNavigate()

  function ButtonClicked(id) {
    navigate("/players/instance?id=" + id)
  }

  return (
    <Card
    style= {{backgroundColor: 'lightgray', minWidth: '100%'}}>
      <Card.Img variant="top" src={headshot} style={{width: '100%', height: '20vw', objectFit: 'cover'}}/>
      <Card.Body>
        <Card.Title style = {{fontSize: '3vh', color: '#333232'}}>{Highlight(name, search)}</Card.Title>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>Team: {Highlight(team, search)}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>Jersey #{Highlight(jersey, search)}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>League: {Highlight(league, search)}</Card.Text>
        <Card.Text style = {{fontSize: '2vh', color: '#404040'}}>College: {Highlight(college, search)}</Card.Text>
        
      </Card.Body>
      <Card.Footer style= {{backgroundColor: 'dimgray'}}>
        <Button class="btn btn-primary stretched-link" onClick={() => ButtonClicked(id)}
        style= {{backgroundColor: '#3d405b', borderColor: 'black'}}>
          More Info
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default PlayerCard;