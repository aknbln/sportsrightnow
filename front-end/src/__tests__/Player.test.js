// Replace useNavigates with a dummy function as it doesn't work without a <Router>

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

import renderer from 'react-test-renderer'
import PlayerCard from '../components/PlayerCard'
import img from '../assets/images/question.png'
import { inchToFeet } from '../Utils'

const Player = 
{
    name: "Bobby Basketball",
    image: img,
    team: "Buffalo Ballers",
    league: "NBA",
    age: "100",
    country: "USA",
    height: "12' 3\"",
    player_id: "999",

}

it('PlayerCard data correct', () => {
    const component = renderer.create(<PlayerCard playerData={Player}/>)
    expect(component.age = 100)
})

it('PlayerCard renders correctly', () => {
    const component = renderer.create(<PlayerCard playerData={Player}/>)
    let json = component.toJSON()
    expect(json).toMatchSnapshot()
})

it('Inch to feet conversion correct', () => {
    const inches = 75
    let feet = inchToFeet(inches)
    expect(feet = "7' 3\"")
})