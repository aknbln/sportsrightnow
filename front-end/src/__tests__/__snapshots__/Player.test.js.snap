// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerCard renders correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "backgroundColor": "lightgray",
    }
  }
>
  <img
    className="card-img-top"
    src="question.png"
    style={
      Object {
        "height": "20vw",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
      style={
        Object {
          "color": "#333232",
          "fontSize": "3vh",
        }
      }
    >
      Bobby Basketball
    </div>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      Team: 
      Buffalo Ballers
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      League: 
      NBA
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      Age: 
      100
    </p>
    <p
      className="card-text"
      style={
        Object {
          "color": "#404040",
          "fontSize": "2vh",
        }
      }
    >
      Nationality: 
      USA
    </p>
  </div>
  <div
    className="card-footer"
    style={
      Object {
        "backgroundColor": "dimgray",
      }
    }
  >
    <button
      class="btn btn-primary stretched-link"
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#3d405b",
          "borderColor": "black",
        }
      }
      type="button"
    >
      More Info
    </button>
  </div>
</div>
`;
